test/python/modules/common/gmm_tests.py

```diff
@@ -0,0 +1,169 @@
+import unittest
+import numpy as np
+import sys
+import os
+
+# root directory of the whole project
+ROOT = os.path.abspath(os.path.join(
```

Comment:
This looks very fragile.  Is there some way around needing it?  If it's necessary can we perhaps pass it in when the tool is called?

Reply:
This script at least need to know where the project root is. So, if you don't like this variant of getting the root, then, maybe, I can make script get it from the cmd args. What do you think?

Reply:
I wouldn't call this any more fragile than any other case of referencing a file using a path relative to the current file.
E.g. including a header in a c++ file. Moving this parameter to cmd args would probably be just as fragile, because we'll have to set this command argument in the `CMakeLists.txt` afterwards. But we'll lose the ability to easily run and debug this script from IDEs.

Reply:
Referencing a header in a C++ file is a little different because it happens at compile time and is configurable outside the file by passing `-I` to the compiler.  Nonetheless you all are more familiar with the system that I am so I will leave the decision to you.

Reply:
(What I'm trying to avoid is having to make changes to a file just because it was moved to a different directory)